code/modules/mob/living/carbon/carbon.dm

```diff
@@ -827,6 +827,22 @@
 	if(!getorgan(/obj/item/organ/brain) && (!mind || !mind.has_antag_datum(/datum/antagonist/changeling)))
 		return 0
 
+/mob/living/carbon/proc/can_defib() //yogs start
+	if(suiciding || hellbound || HAS_TRAIT(src, TRAIT_HUSK)) //can't revive
+		return FALSE
+	if((world.time - timeofdeath) > DEFIB_TIME_LIMIT * 10) //too late
+		return FALSE
+	if((getBruteLoss() >= MAX_REVIVE_BRUTE_DAMAGE) || (getFireLoss() >= MAX_REVIVE_FIRE_DAMAGE) || !can_be_revived()) //too damaged
+		return FALSE
+	if(!getorgan(/obj/item/organ/heart)) //what are we even shocking
+		return FALSE
+	var/obj/item/organ/brain/BR = getorgan(/obj/item/organ/brain)
+	if(QDELETED(BR) || BR.brain_death || BR.damaged_brain || BR.suicided)
+		return FALSE
+	if(get_ghost())
```

Comment:
I'm curious about get_ghost(), won't mean that if the person has disconnected this would still return true?

Reply:
As I remember braindeads can still be defibbed with current defibrillators, and this doesnt work any different.